### DISCUSSION TITLE
DROP TABLE in migration

### DISCUSSION THREAD

**Discussion by Liooo** [ent/ent#3134] [Q&A] [answered]
Currently versioned migration doesn't generate DROP TABLE statements, nor is there `migrate.WithDropTables(true)` option. 
Was there any reasons behind this?

**Comment by godcong** [comment, marked as answer]
How do you determine which tables are to be deleted?
If you are using multiple sql tools, some are generated by ent.
Some are generated by other tools or non-tools.
How do you determine which tables need to be deleted if you simply `drop=true`?

The only solution I can think of is to introduce another table dedicated to migration to manipulate these

**Comment by Liooo** [comment]
Thanks, that should be the reason it doesn't exist.

It'd be nice if there was an option to simply delete all tables that are not ent managed or let it accept whitelist tables, which should be enough to cover the cases that the database includes only ent managed tables, which in my complete guess shouldn't be a rare one.
